PostHog/PostHogRemoteConfig.swift

```diff
@@ -31,9 +45,85 @@ class PostHogFeatureFlags {
         self.storage = storage
         self.api = api
 
+        preloadRemoteConfig()
+        preloadFeatureFlags()
         preloadSessionReplayFlag()
     }
 
+    private func preloadRemoteConfig() {
+        remoteConfigLock.withLock {
+            // load disk cached config to memory
+            _ = getCachedRemoteConfig()
+        }
+
+        // may have already beed fetched from `loadFeatureFlags` call
+        if remoteConfigLock.withLock({
+            self.remoteConfig == nil || !self.remoteConfigDidFetch
+        }) {
+            dispatchQueue.async {
+                self.reloadRemoteConfig()
+            }
+        }
+    }
+
+    private func preloadFeatureFlags() {
+        featureFlagsLock.withLock {
+            // load disk cached config to memory
+            _ = getCachedFeatureFlags()
+        }
+
+        if config.preloadFeatureFlags {
```

Comment:
we should add a config for the remote config API as well and fallback to the decide API in case the remote config is disabled

Reply:
same on RN, we just need to add a comment that disabling this would disable surveys.
since this is an experimental feature (remote config), we need a way to disable it and fallback to decide API for everything else